src/main/java/org/candlepin/resteasy/filter/SecurityHoleAuthorizationFilter.java

```diff
@@ -61,13 +89,43 @@ void runFilter(ContainerRequestContext requestContext) {
 
         Principal principal = (Principal) requestContext.getSecurityContext().getUserPrincipal();
 
+        ResourceInfo resourceInfo = ResteasyContext.getContextData(ResourceInfo.class);
+        Method method = resourceInfo.getResourceMethod();
+        SecurityHole securityHole = annotationLocator.getAnnotation(method, SecurityHole.class);
+
         if (principal instanceof ActivationKeyPrincipal) {
-            ResourceInfo resourceInfo = ResteasyContext.getContextData(ResourceInfo.class);
-            Method method = resourceInfo.getResourceMethod();
-            SecurityHole securityHole = annotationLocator.getAnnotation(method, SecurityHole.class);
             if (!securityHole.activationKey()) {
                 denyAccess(principal, method);
             }
         }
+        if (principal instanceof AnonymousCloudConsumerPrincipal anonymPrincipal) {
+            if (!securityHole.anonConsumer()) {
+                denyAccess(principal, method);
+            }
+            AnonymousCloudConsumer anonymousCloudConsumer = anonymPrincipal.getAnonymousCloudConsumer();
+            String ownerKey = checkCloudAccountOrganization(anonymousCloudConsumer);
+            if (!ownerCurator.existsByKey(ownerKey)) {
+                throw new TooManyRequestsException(ORG_NOT_CREATED_IN_CANDLEPIN);
+            }
+            if (!poolCurator.hasPoolsForProducts(ownerKey, anonymousCloudConsumer.getProductIds())) {
+                throw new TooManyRequestsException(ORG_DOES_NOT_HAVE_POOLS);
+            }
+            anonymousCloudConsumer.setOwnerKey(ownerKey);
+        }
+    }
+
+    private String checkCloudAccountOrganization(AnonymousCloudConsumer anonymousCloudConsumer) {
+        try {
+            return cloudAdapter.checkCloudAccountOrgIsReady(
+                anonymousCloudConsumer.getCloudAccountId(),
+                anonymousCloudConsumer.getCloudProviderShortName(),
+                anonymousCloudConsumer.getCloudOfferingId());
+        }
+        catch (OrgForCloudAccountNotCreatedYetException e) {
+            throw new TooManyRequestsException(ORG_DOES_NOT_EXIST_AT_ALL);
+        }
+        catch (CouldNotEntitleOrganizationException e) {
```

Comment:
Need to update `CouldNotEntitleOrganizationException` to `OrgForCloudAccountNotEntitledYetException`

Reply:
done